source/material_model/viscoelastic.cc

```diff
@@ -0,0 +1,696 @@
+      return average_viscoelastic_viscosity;
+    }
+
+
+    template <int dim>
+    void
+    Viscoelastic<dim>::
+    evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+             MaterialModel::MaterialModelOutputs<dim> &out) const
+    {
+
+
+      // Check whether the compositional fields representing the viscoelastic
+      // stress tensor are both named correctly and listed in the right order.
+      if ( dim == 2)
+        {
+          AssertThrow(this->introspection().compositional_index_for_name("stress_xx") == 0,
+                      ExcMessage("Material model Viscoelastic only works if the first "
+                                 "compositional field is called stress_xx."));
+          AssertThrow(this->introspection().compositional_index_for_name("stress_yy") == 1,
+                      ExcMessage("Material model Viscoelastic only works if the second "
+                                 "compositional field is called stress_yy."));
+          AssertThrow(this->introspection().compositional_index_for_name("stress_xy") == 2,
+                      ExcMessage("Material model Viscoelastic only works if the third "
+                                 "compositional field is called stress_xy."));
+        }
+      if ( dim == 3)
+        {
+          AssertThrow(this->introspection().compositional_index_for_name("stress_xx") == 0,
+                      ExcMessage("Material model Viscoelastic only works if the first "
+                                 "compositional field is called stress_xx."));
+          AssertThrow(this->introspection().compositional_index_for_name("stress_yy") == 1,
+                      ExcMessage("Material model Viscoelastic only works if the second "
+                                 "compositional field is called stress_yy."));
+          AssertThrow(this->introspection().compositional_index_for_name("stress_zz") == 2,
+                      ExcMessage("Material model Viscoelastic only works if the third "
+                                 "compositional field is called stress_zz."));
+          AssertThrow(this->introspection().compositional_index_for_name("stress_xy") == 3,
+                      ExcMessage("Material model Viscoelastic only works if the fourth "
+                                 "compositional field is called stress_xy."));
+          AssertThrow(this->introspection().compositional_index_for_name("stress_xz") == 4,
+                      ExcMessage("Material model Viscoelastic only works if the fifth "
+                                 "compositional field is called stress_xz."));
+          AssertThrow(this->introspection().compositional_index_for_name("stress_yz") == 5,
+                      ExcMessage("Material model Viscoelastic only works if the sixth "
+                                 "compositional field is called stress_yz."));
```

Comment:
Please move all of these asserts into `parse_parameters` or `initialize`. We do not need to check this every time the model is called.